src/MultiPrecisionR2.jl

```diff
@@ -372,6 +372,8 @@ function SolverCore.solve!(
     )
   end
 
+  callback()
```

Comment:
I suggest something like
callback = (args...) -> nothing,

so it is more general with inputs 
We need it in the argument of function

Reply:
We need to have something like this callback(nlp, solver, stats)
so we send stats back to the callback and nlp , solver is not important now but in future would be useful


we also need to call cb before the loop and in the loop

Reply:
I do not think we need args.. as input of the solver, they are splat operator https://docs.julialang.org/en/v1/manual/functions/#Varargs-Functions